src/transformers/cache_utils.py

```diff
@@ -2073,375 +1793,632 @@ def _prefetch_layer_in_context(self, layer_idx: int) -> None:
             self.device_value_cache[self.active_device_layer].fill_(0.0)
 
 
-class MambaCache:
+class OffloadedStaticCache(StaticCache):
     """
-    Cache for mamba model which does not have attention mechanism and key value states.
+    A drop-in replacement for StaticCache that conserves accelerator memory by offloading
+    cache tensors to CPU when not actively being used.
 
-    Arguments:
-        config (`PretrainedConfig):
-            The configuration file defining the shape-related attributes required to initialize the static cache.
-        max_batch_size (`int`):
-            The maximum batch size with which the model will be used. Note that a new instance must be instantiated if a smaller batch size is used.
-        dtype (`torch.dtype`, *optional*, defaults to `torch.float16`):
-            The default `dtype` to use when initializing the layer.
-        device (`torch.device` or `str`, *optional*):
-            The device on which the cache should be initialized. Should be the same as the layer.
+    This cache maintains the compilation-friendly properties of StaticCache while enabling
+    much longer sequences by offloading inactive layers to CPU memory.
+
+    Parameters:
+        config (`PretrainedConfig`): Model configuration for shape/device info.
+        max_batch_size (`int`): Maximum batch size for static caches.
+        max_cache_len (`int`, *optional*): Maximum sequence length.
+        device (`torch.device` or `str`, *optional*): Device for cache tensors.
+        dtype (`torch.dtype`, *optional*, defaults to `torch.float32`): Data type for cache tensors.
+        offload_device (`Union`, *optional*, defaults to `"cpu"`): Device to offload cache tensors to.
+        layer_device_map (`dict[int, Union[str, torch.device, int]]`, *optional*): Per-layer device mapping.
 
     Example:
+        ```python
+        >>> from transformers import AutoTokenizer, AutoModelForCausalLM, OffloadedStaticCache
+
+        >>> model = AutoModelForCausalLM.from_pretrained("openai-community/gpt2")
+        >>> tokenizer = AutoTokenizer.from_pretrained("openai-community/gpt2")
+
+        >>> inputs = tokenizer(text="My name is GPT2", return_tensors="pt")
+
+        >>> # Prepare a cache class with offloading
+        >>> max_generated_length = inputs.input_ids.shape[1] + 10
+        >>> past_key_values = OffloadedStaticCache(
+        ...     config=model.config,
+        ...     max_batch_size=1,
+        ...     max_cache_len=max_generated_length,
+        ...     device=model.device,
+        ...     dtype=model.dtype
+        ... )
+        >>> outputs = model(**inputs, past_key_values=past_key_values, use_cache=True)
+        >>> outputs.past_key_values # access cache with offloaded layers
+        OffloadedStaticCache()
+        ```
+    """
+
+    def __init__(
```

Comment:
good for BC, but maybe we want to move forward with just specifying `"cache_processor="offloaded"` wdyt @Cyrilvallez